consensus/core/src/network/mod.rs

```diff
@@ -44,13 +45,15 @@ pub(crate) trait NetworkClient: Send + Sync + 'static {
         timeout: Duration,
     ) -> ConsensusResult<()>;
 
-    /// Fetches serialized `SignedBlock`s from a peer.
+    /// Fetches serialized `SignedBlock`s from a peer. It also might return additional ancestor blocks
+    /// of the requested blocks according to the provided `highest_accepted_rounds`.
     async fn fetch_blocks(
         &self,
         peer: AuthorityIndex,
         block_refs: Vec<BlockRef>,
+        highest_accepted_rounds: Vec<Round>,
         timeout: Duration,
-    ) -> ConsensusResult<Vec<Bytes>>;
+    ) -> ConsensusResult<(Vec<Bytes>, Vec<Bytes>)>;
```

Comment:
That's a good point, but I didn't want to do that to avoid leaking the `FetchBlocksResponse` outside of the network module. If we think this is ok then I can certainly do.